src/rust/e2e/src/market_registration.rs

```diff
@@ -0,0 +1,45 @@
+use anyhow::{ensure, Result};
+use aptos_sdk::types::APTOS_COIN_TYPE;
+use e2e_proc_macro::e2e_test;
+use econia_sdk::entry::register_market_base_coin_from_coinstore;
+use sqlx::query;
+
+use metadata::Metadata;
+
+use crate::utils::*;
+
+#[e2e_test]
+pub async fn test_market_registration<'a>(state: &'a State) -> Result<()> {
+    let lot_size = 10u64.pow(8 - 3); // eAPT has 8 decimals, want 1/1000th granularity
+    let tick_size = 10u64.pow(6 - 3); // eAPT has 6 decimals, want 1/1000th granularity
+    let min_size = 1;
+
+    let entry = register_market_base_coin_from_coinstore(
+        state.econia_address.clone(),
+        &state.e_apt,
+        &state.e_usdc,
+        &APTOS_COIN_TYPE,
+        lot_size,
+        tick_size,
+        min_size,
+    )
+    .unwrap();
+
+    let (_, mut econia_client) =
+        account(&state.faucet_client, &state.node_url, state.econia_address).await;
+
+    econia_client.submit_tx(entry).await?;
+
+    std::thread::sleep(TIMEOUT);
+
+    let result = query!("SELECT COUNT(*) AS count FROM market_registration_events")
```

Comment:
Isn't there a delay interval on all the aggregations, so this might (will probably) spuriously fail?
Probably need to be mindful of the delays and wait twice as long to give the aggregator time.

Reply:
I would suggest having a very small timeout on the aggregator while testing when possible.